### DISCUSSION TITLE
Modify how Better BibTeX handles key clashes

### DISCUSSION THREAD

**Discussion by bwolleswinkel** [retorquere/zotero-better-bibtex#2973] [Q&A] [answered]
My citation key formula is 'auth.lower + shorttitle(3,3) + year', and sometimes key clashes occur. Now, BBT handles this by appending a postfix 'a' to one of the keys (preventing the clash), but I'd like it to be smarter (is this possible?).

The first reason a clash occurs could be that two papers (by the same author, same year) are distinct but have identical titles up to the first 3 words (e.g., 'Observer based self-triggered control of linear plants with unknown disturbances' and 'Observer based self-triggered control of an acyclic interconnection of linear plants'). In that case, would it be possible to tell BBT to use the citation key formula 'auth.lower + shorttitle(x,x) + year', where it keeps increasing x > 3 until a collision no longer occurs?

The second reason a clash can occur is that there exists a preprint and journal version of a reference (with identical titles, published in the same year). As the former will fail, would it then instead be possible to tell BBT to use would it be possible to tell BBT to use the citation key formula 'auth.lower + shorttitle(x,x) + year + type'?

**Comment by retorquere** [comment, marked as answer]
> would it be possible to tell BBT to use the citation key formula 'auth.lower + shorttitle(x,x) + year', where it keeps increasing x > 3 until a collision no longer occurs?

Not realistically, no. I can explain why if you want but this would make the key generator way more complex than it already is.

> The second reason a clash can occur is that there exists a preprint and journal version of a reference

Not in a general way, no, for the same reasons. In this specific case you can do:

```
type(preprint) + auth.lower + shorttitle(x,x) + year + type | auth.lower + shorttitle(x,x) + year
```

**Reply by bwolleswinkel** [reply]
Thanks for the quick response and the suggestion! I was counting on there being no easy fix, but again, the tool works great even without, so I'll leave it be then :)
